src/VisualStudio/Core/Def/Implementation/TableDataSource/VisualStudioBaseDiagnosticListTable.LiveTableDataSource.cs

```diff
@@ -391,8 +391,23 @@ private ErrorRank GetErrorRank(DiagnosticData item)
                     switch (value)
                     {
                         case WellKnownDiagnosticTags.Build:
-                            // any error from build is highest priority
-                            return ErrorRank.Lexical;
+                            // any error from build gets lowest priority
+                            // see https://github.com/dotnet/roslyn/issues/28807
+                            //
+                            // this is only used when intellisense (live) errors are involved.
+                            // with "build only" filter on, we use order of errors came in from build for ordering
+                            // and doesn't use ErrorRank for ordering (by giving same rank for all errors)
+                            //
+                            // when live errors are involved, by default, error list will use the following to sort errors
+                            // error rank > project rank > project name > file name > line > column
+                            // which will basically make syntax errors show up before declaration error and method body semantic errors
+                            // among same type of errors, leaf project's error will show up first and then projects that depends on the leaf projects
+                            //
+                            // any build errors mixed with live errors will show up at the end. when live errors are on, some of errors
+                            // still left as build errors such as errors produced after CompilationStages.Compile or ones listed here
+                            // http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/Compilation/CSharpCompilerDiagnosticAnalyzer.cs,23 or similar ones for VB
+                            // and etc.
+                            return ErrorRank.PostBuild;
```

Comment:
I don't know how this works so just want to understand; if there's a lexical "build" error and it's not caught by "live" - shouldn't these appear at the same rank as the other lexical errors "ErrorRank.Lexical". Same with semantic, emit, etc?

Reply:
we don't know what kind of errors it is if it came from build. if we could know, merging would have been much easier.